### PR TITLE
feat: upgrade support to mongodb v7

### PR DESCRIPTION
**Pull request by caioq**
## Why

The current version of mongoose (v5) doesn't support the latest version of mongodb.

## What

- Upgrade mongooose to v8
- Upgrade typescript to v5
- Upgrade node to v18

## Tests
Run locally:
image

image

### CODE REVIEW

**Comment by caioq** [comment]
@SvanBoxel can you help me with a review?

**Review by SvanBoxel — APPROVED** [review]

**Review by SvanBoxel — CHANGES_REQUESTED** [review]
Could you undo all the linting changes and/or make them part of a separate PR?

**Comment by caioq** [comment]
> Could you undo all the linting changes and/or make them part of a separate PR?

Sure! I undid some format linting changes, is that your request?
I had to upgrade the eslint, node and typescript dependencies because some conflict with the new version of the mongoose dependency.

**Comment by SvanBoxel** [comment]
Hey, I still see a lot of formatting changes. I hope we can keep the scope of this PR to the upgrade of the packages without changing the rules that are applied to the code.

**Comment by caioq** [comment]
> Hey, I still see a lot of formatting changes. I hope we can keep the scope of this PR to the upgrade of the packages without changing the rules that are applied to the code.

Hi! I had a hard time undoing some formatting changes due to different rules applied to parts of the code, but now I think I managed it completely. 
Then I can open a new PR to establish a formatting rule throughout the code base

File: src/handlers/re-run.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ import { Context } from 'probot' // eslint-disable-line @typescript-eslint/no-un
 import Runs from '../models/runs.model'
 
 async function handleReRun(context: Context): Promise<void> {
+  console.debug('Handling Re run event')
```
Comment (SvanBoxel) [inline src/handlers/re-run.ts]: Do we need this console.debug as part of this PR?

File: src/handlers/push.ts [hunk 2]
```diff
@@ -68,6 +69,7 @@ async function handlePush(context: Context): Promise<void> {
       event: context.payload
     }
   })
+  console.debug('Event dispatched', config.workflows_repository, JSON.stringify(data), JSON.stringify(response))
```
Comment (SvanBoxel) [inline src/handlers/push.ts]: Do we need this console.debug as part of this PR?

File: src/handlers/push.ts [hunk 3]
```diff
@@ -2,13 +2,14 @@ import { Context } from 'probot' // eslint-disable-line @typescript-eslint/no-un
...
 import Run from '../models/runs.model'
-import { default_organization_repository, app_route, config_keys } from "../constants";
+import { default_organization_repository, app_route, config_keys, repository_dispatch_type } from "../constants";
 import shouldRun from "../utils/should-run";
 
-export const repository_dispatch_type = 'org-workflow-bot'
 export const config_path = 'organization-workflows-settings.yml'
 
 async function handlePush(context: Context): Promise<void> {
+  console.debug('Handling push event', context.payload.repository.name)
```
Comment (SvanBoxel) [inline src/handlers/push.ts]: Do we need this console.debug as part of this PR?

**Comment by SvanBoxel** [comment]
Thank you, that already makes reviewing a bit easier. I also noticed you added a bunch of `console.debug` statements? Are those needed as part of this PR?

**Comment by caioq** [comment]
> Thank you, that already makes reviewing a bit easier. I also noticed you added a bunch of `console.debug` statements? Are those needed as part of this PR?

I added it to make debugging the application easier. I can remove them if you think it's better.

**Comment by SvanBoxel** [comment]
For this PR, let's limit it to upgrading the packages. If you think other improvements can be made, it's best to propose those in separate PRs. Besides that it looks good to me! Thank you for working on this.

**Comment by caioq** [comment]
> For this PR, let's limit it to upgrading the packages. If you think other improvements can be made, it's best to propose those in separate PRs. Besides that it looks good to me! Thank you for working on this.

Thanks! 
I have removed the debug logs here https://github.com/SvanBoxel/organization-workflows/pull/51/commits/04c61ac5bed5bfbded473196fc541e7d4c472447

**Review by SvanBoxel — APPROVED** [review]

**Comment by SvanBoxel** [comment]
Published as pre-release: https://github.com/SvanBoxel/organization-workflows/releases/tag/1.5.0

**Pull request merged**